tox.ini

```diff
@@ -17,9 +17,8 @@ deps =
     #mitmproxy >= 5.3.0; python_version >= '3.9' and implementation_name != 'pypy'
     # The tests hang with mitmproxy 8.0.0: https://github.com/scrapy/scrapy/issues/5454
     mitmproxy >= 4.0.4, < 8; python_version >= '3.7' and python_version < '3.9' and implementation_name != 'pypy'
-    mitmproxy >= 4.0.4, < 5; python_version >= '3.6' and python_version < '3.7' and platform_system != 'Windows' and implementation_name != 'pypy'
     # newer markupsafe is incompatible with deps of old mitmproxy (which we get on Python 3.7 and lower)
-    markupsafe < 2.1.0; python_version >= '3.6' and python_version < '3.8' and implementation_name != 'pypy'
+    markupsafe < 2.1.0; python_version >= '3.7' and python_version < '3.8' and implementation_name != 'pypy'
```

Comment:
- `# mitmproxy does not support Windows when running Python < 3.7` can be removed now
- `mitmproxy >= 4.0.4, < 8; python_version >= '3.7' and python_version < '3.9' and implementation_name != 'pypy'` can be simplified as `mitmproxy >= 4.0.4, < 8; python_version < '3.9' and implementation_name != 'pypy'`
- `markupsafe < 2.1.0; python_version >= '3.7' and python_version < '3.8' and implementation_name != 'pypy'` can be simplified as `markupsafe < 2.1.0; python_version < '3.8' and implementation_name != 'pypy'` (it could already before your change, [an oversight from me during the last update](https://github.com/scrapy/scrapy/pull/4742/files#diff-6aa3751e2010724eb62472e6ebbb0aa229da6a9e10678a53b6e6373a40614e88))